packages/flutter/lib/src/material/search.dart

```diff
@@ -247,10 +247,13 @@ abstract class SearchDelegate<T> {
   ///
   /// If the user taps on a suggestion provided by [buildSuggestions] this
   /// string should be updated to that suggestion via the setter.
+  ///
+  /// As soon as the query is set, the cursor moves to the end position
   String get query => _queryTextController.text;
   set query(String value) {
```

Comment:
actually it might be best to put the sentence you're adding here, as dartdoc to the setter. Something like:
```
/// Changes the current query string.
///
/// Setting the query string programmatically moves the cursor to the end of the text field.
set query(...) {
```